cli/admin/src/main/java/io/pravega/cli/admin/controller/metadata/ControllerMetadataUpdateEntryCommand.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.controller.metadata;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.cli.admin.json.ControllerMetadataJsonSerializer;
+import io.pravega.cli.admin.serializers.controller.ControllerMetadataSerializer;
+import io.pravega.cli.admin.utils.AdminSegmentHelper;
+import io.pravega.client.tables.impl.TableSegmentEntry;
+import io.pravega.client.tables.impl.TableSegmentKeyVersion;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Paths;
+
+public class ControllerMetadataUpdateEntryCommand extends ControllerMetadataCommand {
+
+    /**
+     * Creates a new instance of the Command class.
+     *
+     * @param args The arguments for the command.
+     */
+    public ControllerMetadataUpdateEntryCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(4);
+
+        final String tableName = getArg(0);
+        final String key = getArg(1);
+        final String newValueFile = getArg(2);
+        final String segmentStoreHost = getArg(3);
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        AdminSegmentHelper adminSegmentHelper = instantiateAdminSegmentHelper(zkClient);
+        ControllerMetadataSerializer serializer = new ControllerMetadataSerializer(tableName, key);
+        ControllerMetadataJsonSerializer jsonSerializer = new ControllerMetadataJsonSerializer();
+
+        String jsonValue;
+        try {
+            jsonValue = new String(Files.readAllBytes(Paths.get(newValueFile)));
+        } catch (NoSuchFileException e) {
+            output("File with new value does not exist: %s", newValueFile);
+            return;
+        }
+        ByteBuffer updatedValue = serializer.serialize(jsonSerializer.fromJson(jsonValue, serializer.getMetadataClass()));
```

Comment:
Please also test for negative test scenarios where the jsonValue supplied by the User is malformed or not as per expectation... I would expect to see a JsonParseException thrown at this line... but it would be good to test and verify.

Reply:
I've added a test for the any JSON exception cases.

Reply:
Cool Thanks.